docs/usage/key-concepts/changelogs.md

```diff
@@ -0,0 +1,109 @@
+1. Looking up the source URL, if it resides on a supported platform (e.g. GitHub)
+1. Checking for both "Releases" metadata in the repository and any commonly known "changelog" file names
+1. Filtering the found releases to only include those versions being updated by the current PR
+1. Formatting and embedding the results into the PR body
+
+## Changelogs for private packages
+
+For private packages, the algorithm is mostly the same as described above, with the additional considerations:
+
+- Renovate must be able to access the private package in the first place
+- The private registry must include the source URL in its response
+- Renovate must be able to detect and authenticate with whatever private repository corresponds to the source URL
+
+For more details, see [Private packages, looking up changelogs](../getting-started/private-packages.md#looking-up-changelogs).
+
+## Relevant configuration options
+
+### [`fetchChangelogs`](../configuration-options.md#fetchchangelogs)
+
+Set to `off` if changelog fetching is causing a problem.
+
+Set to `branch` if you have an advanced use case where you're embedding changelogs in the Git commit itself, we don't recommend this due to its potential size.
+
+### [`customChangelogUrl`](../configuration-options.md#customchangelogurl)
+
+This doesn't help with _fetching_ the changelogs, but if you configure it then Renovate will include a link to this URL in the PR body, so users can click through to read the changelog.
+
+## Platforms that Renovate can fetch changelogs from
+
+See the list of platforms in the [`fetchChangelogs` config option docs](../configuration-options.md#fetchchangelogs).
+
+### Running Renovate on a non-GitHub platform
+
+Most Open Source packages are hosted on github.com, which means most changelogs are hosted there too.
+Fetching changelogs from github.com requires a GitHub token because GitHub blocks unauthenticated GraphQL API use.
+
+This means that if you run Renovate on self-hosted GitHub Enterprise Server, or any non-GitHub platform which Renovate supports, then you need to configure a github.com Personal Access Token in Renovate in order to fetch changelogs.
+
+Read [Running Renovate, GitHub.com token for changelogs](../getting-started/running.md#githubcom-token-for-changelogs) to learn more.
+
+## Troubleshooting missing changelogs
+
+Follow these steps to find out why Renovate does not find a changelog:
+
+1. The datasource for this package does not support sourceUrls.
+   - If the registry fundamentally does not provide this data, then the only possibility is for it to be manually populated through PRs to Renovate's source code
+   - If the registry provides source URLs but Renovate does not support it yet, then raise a feature request with examples, or better yet a Pull Request to implement it yourself
```

Comment:
What does `Renovate does not support it yet,` mean here? Same question with `a Pull Request to implement it yourself`.